protocol/contracts/leaser/src/migrate.rs

```diff
@@ -36,12 +36,12 @@ where
     Leases: ExactSizeIterator<Item = Addr>,
     I: Iterator<Item = MaybeCustomer<LI>>,
     LI: ExactSizeIterator<Item = Addr>,
-    MsgFactory: Fn() -> MigrateMsg,
+    MsgFactory: Fn(MigrationKind) -> MigrateMsg,
 {
     let mut msgs = MigrateBatch::new(lease_code, max_leases);
 
     let legacy_leases = msgs
-        .migrate_leases(leases, &migrate_msg)
+        .migrate_leases(leases, &|| migrate_msg(MigrationKind::LEGACY))
```

Comment:
Taking a reference to the closure is most likely unnecessary and might also be caught by clippy in the CI.

Reply:
neither of them - the compiler complains if not provided, and the clippy is happy with it